backend/app/Services/Traceability/TraceabilityService.php

```
@@ -55,14 +58,65 @@ public function forwardTrace(MaterialLot $lot): array
+     *
+     * @return array{pallets: \Illuminate\Support\Collection, customer_orders: \Illuminate\Support\Collection}
+     */
+    private function finishedGoodsPacking(MaterialLot $lot): array
+    {
+        if ($lot->source_batch_id === null) {
+            return ['pallets' => collect(), 'customer_orders' => collect()];
+        }
+
+        // withTrashed: a finished lot must still resolve its pallets / customer
+        // order if the producing batch was later soft-deleted (recall readiness).
+        $pallets = Pallet::where('batch_id', $lot->source_batch_id)
```

PHPStan (2.2.2)

[error] Call to an undefined static method App\Models\Pallet::where().


(staticMethod.notFound)